File: exercises/py101/easy1_small_problems/short_long_short.py


# returns short long short
def short_long_short(sentence1, sentence2):
    # assume the strings are of different lengths
    long = sentence1 if len(sentence1) > len(sentence2) else sentence2
    short = sentence2 if len(sentence1) > len(sentence2) else sentence1 
    return f"{short}{long}{short}"

# These examples should all print True
print(short_long_short('abc', 'defgh') == "abcdefghabc")
print(short_long_short('abcde', 'fgh') == "fghabcdefgh")
print(short_long_short('', 'xyz') == "xyz")
    

File: exercises/js210/easy1/short_long_short.js
function generateString(shortStr, longStr) {
  return shortStr + longStr + shortStr;
}

function shortLongShort(str1, str2) {
  let result = '';
  if (str1.length > str2.length) {
    result = generateString(str2, str1);
  } else {
    result = generateString(str1, str2);
  }
  return result;
}

console.log(shortLongShort('abc', 'defgh'));    // "abcdefghabc"
console.log(shortLongShort('abcde', 'fgh'));    // "fghabcdefgh"
console.log(shortLongShort('', 'xyz'));         // "xyz"